src/3d/qgspointcloudlayerchunkloader_p.cpp

```diff
@@ -175,13 +175,28 @@ QVector<QgsChunkNode *> QgsPointCloudLayerChunkLoaderFactory::createChildren( Qg
   const float childError = node->error() / 2;
   float xc = bbox.xCenter(), yc = bbox.yCenter(), zc = bbox.zCenter();
 
+  QgsRectangle extent;
+  try
+  {
+    extent = mCoordinateTransform.transformBoundingBox( mMap.extent(), Qgis::TransformDirection::Reverse );
+  }
+  catch ( const QgsCsException & )
+  {
+    // bad luck, can't reproject for some reason
+    QgsDebugMsg( QStringLiteral( "Transformation of extent failed." ) );
+  }
```

Comment:
Rather than doing the transform every time we create children, how about we transform it once when the chunk loader is created, and store the extent in layer CRS as a member variable?

Reply:
this makes more sense, yes!